climpred/prediction.py

```diff
@@ -340,30 +351,25 @@ def compute_persistence(
         n, freq = get_lead_cftime_shift_args(hind.lead.attrs['units'], 1)
         # Shift backwards shift for lead zero.
         hind['init'] = shift_cftime_index(hind, 'init', -1 * n, freq)
-    nlags = max(hind.lead.values)
     # temporarily change `init` to `time` for comparison to verification data time.
     hind = hind.rename({'init': 'time'})
-    if alignment != 'maximize':
-        # slices down to inits in common with hindcast, plus gives enough room
-        # for maximum lead time forecast.
-        a, _ = reduce_time_series_for_aligned_inits(hind, verif, nlags)
-        inits = a['time']
+    if alignment == 'same_inits':
+        a, _ = reduce_forecast_to_same_inits(hind, verif)
+        initial = verif.sel(time=a['time'])
+    else:
+        raise NotImplementedError(
+            "Only 'same_inits' alignment is currently implemented."
+        )
 
     plag = []
     for lag in hind.lead.values:
-        if alignment == 'maximize':
-            # slices down to inits in common with hindcast, but only gives enough
-            # room for lead from current forecast
-            a, _ = reduce_time_series_for_aligned_inits(hind, verif, lag)
-            inits = a['time']
         n, freq = get_lead_cftime_shift_args(hind.lead.attrs['units'], lag)
         target_dates = shift_cftime_index(a, 'time', n, freq)
 
         o = verif.sel(time=target_dates)
-        f = verif.sel(time=inits)
-        o['time'] = f['time']
+        o['time'] = initial['time']
```

Comment:
what is `initial` when `alignment=maximize`?

Reply:
I just wrote this with `same_inits` in mind. I'll probably change the variable name to something else with the other alignment types. `initial` in the case of 'maximize' here would just get overwritten on each loop. I bumped it out of the loop for this case, since it doesn't need to be calculated every loop.